grr/server/grr_response_server/flows/general/osquery.py

```diff
@@ -85,8 +200,51 @@ def Process(
 
     self._UpdateProgress(responses)
 
+    # Make Osquery results available before files are collected
     for response in responses:
       self.SendReply(response)
 
+    self._FileCollectionFromColumns(responses)
+
   def GetProgress(self) -> rdf_osquery.OsqueryProgress:
     return self.state.progress
+
+  def ReceiveFetchedFile(self,
+                         stat_entry,
+                         file_hash,
+                         request_data=None,
+                         is_duplicate=False) -> None:
+    self.SendReply(stat_entry)
+
+  def GetFilesArchiveMappings(
+      self,
+      flow_results: Iterator[rdf_flow_objects.FlowResult],
+  ) -> Iterator[flow_base.ClientPathArchiveMapping]:
+    def _GetUniqueSuffixFor(target_path_so_far: str) -> str:
+      times_used_before = self.state.path_to_count.get(target_path_so_far, 0)
+      self.state.path_to_count[target_path_so_far] = times_used_before + 1
+
+      if times_used_before > 0:
+        return f"-{times_used_before}"
+      else:
+        return ""
+
+    def GenerateTargetPath(pathspec: rdf_paths.PathSpec) -> str:
+      path_info = rdf_objects.PathInfo.FromPathSpec(pathspec)
+      client_string_path = "/".join(path_info.components)
+
+      target_path = f"osquery_collected_files/{client_string_path}"
+      target_path += _GetUniqueSuffixFor(target_path)
+      return target_path
+
+    for result in flow_results:
+      try:
+        osquery_file = _ExtractFileInfo(result)
+      except _ResultNotRelevantError:
+        continue
+
+      client_path = db.ClientPath.FromPathSpec(
+          self.client_id, osquery_file.stat_entry.pathspec)
+      target_path = GenerateTargetPath(osquery_file.stat_entry.pathspec)
+
+      yield flow_base.ClientPathArchiveMapping(client_path, target_path)
```

Comment:
`ClientPathArchiveMapping` is a named tuple, so constructing it should use named fields.

Reply:
I agree; this should be changed at a few other places too, for example:
https://github.com/google/grr/blob/master/grr/server/grr_response_server/flows/general/webhistory.py#L439